hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +49,362 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
     // sync the embedded group
     m_cl->setEmbeddedGroup(m_embed_group);
 
+// check for collision warnings
+#ifdef ENABLE_HIP
+    if (m_exec_conf->isCUDAEnabled())
+        {
+        checkCollisionWarningsGPU(timestep);
+        }
+    else
+#endif
+        {
+        checkCollisionWarnings(timestep);
+        }
+
```

Comment:
Since this warning check is only expected to run infrequently, I would just write the CPU implementation for now. If we find a case where this really is a slow point, you can add a GPU version.